src/attribute.rs

```diff
@@ -0,0 +1,787 @@
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use strum_macros::{Display, EnumIter, EnumString};
+
+#[derive(Debug, PartialEq)]
+pub enum PacketType {
+    Bootp(BootpAttr),
+    Conn(ConnAttr),
+    Dhcp(DhcpAttr),
+    Dns(DnsAttr),
+    Ftp(FtpAttr),
+    Http(HttpAttr),
+    Kerberos(KerberosAttr),
+    Ldap(LdapAttr),
+    Log(LogAttr),
+    Mqtt(MqttAttr),
+    Network(NetworkAttr),
+    Nfs(NfsAttr),
+    Ntlm(NtlmAttr),
+    Rdp(RdpAttr),
+    Smb(SmbAttr),
+    Smtp(SmtpAttr),
+    Ssh(SshAttr),
+    Tls(TlsAttr),
+    Window(WindowAttr),
+}
+
+impl PacketType {
+    /// Creates a new `PacketType` with the given packet type and attribute name.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if `PacketType` creation fails.
+    pub fn from_type_and_attr_name(packet_type: &str, attr_name: &str) -> Result<PacketType> {
+        macro_rules! handle_attr {
+            ($attr:ident, $type:ident) => {
+                $attr::from_str(attr_name).map(PacketType::$type)
+            };
+        }
+
+        let parse_result = match packet_type {
+            "bootp" => handle_attr!(BootpAttr, Bootp),
+            "conn" => handle_attr!(ConnAttr, Conn),
+            "dhcp" => handle_attr!(DhcpAttr, Dhcp),
+            "dns" => handle_attr!(DnsAttr, Dns),
+            "ftp" => handle_attr!(FtpAttr, Ftp),
+            "http" => handle_attr!(HttpAttr, Http),
+            "kerberos" => handle_attr!(KerberosAttr, Kerberos),
+            "ldap" => handle_attr!(LdapAttr, Ldap),
+            "log" => handle_attr!(LogAttr, Log),
+            "mqtt" => handle_attr!(MqttAttr, Mqtt),
+            "network" => handle_attr!(NetworkAttr, Network),
+            "nfs" => handle_attr!(NfsAttr, Nfs),
+            "ntlm" => handle_attr!(NtlmAttr, Ntlm),
+            "rdp" => handle_attr!(RdpAttr, Rdp),
+            "smb" => handle_attr!(SmbAttr, Smb),
+            "smtp" => handle_attr!(SmtpAttr, Smtp),
+            "ssh" => handle_attr!(SshAttr, Ssh),
+            "tls" => handle_attr!(TlsAttr, Tls),
+            "window" => handle_attr!(WindowAttr, Window),
+            _ => return Err(anyhow!("unknown packet type")),
+        };
+        parse_result.map_err(|e| anyhow!("unknown attribute name: {e}"))
+    }
+}
+
+#[derive(Debug, EnumString, PartialEq, EnumIter, Display)]
+pub enum BootpAttr {
+    #[strum(serialize = "Source IP")]
+    SrcAddr,
+    #[strum(serialize = "Source Port")]
+    SrcPort,
+    #[strum(serialize = "Destination IP")]
+    DstAddr,
+    #[strum(serialize = "Destination Port")]
+    DstPort,
+    #[strum(serialize = "Protocol Number")]
+    Proto,
+    #[strum(serialize = "Operation code")]
```

Comment:
For consistency, isn't `Operation Code` better?

Reply:
done